### PR TITLE
MINOR: Refactor DescribeAuthorizedOperationsTest

### PR DESCRIPTION
**Pull request by dajac**
I was looking at `DescribeAuthorizedOperationsTest` in the context of KIP-700. I have made few changes to improve the readability of the code/tests. The PR only moves code around.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala [hunk 1]
```diff
@@ -86,54 +122,36 @@ class DescribeAuthorizedOperationsTest extends IntegrationTestHarness with SaslS
...
-
-  def createConfig(): Properties = {
+  private def createConfig(): Properties = {
     val adminClientConfig = new Properties()
     adminClientConfig.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList)
     adminClientConfig.put(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG, "20000")
     val securityProps: util.Map[Object, Object] =
       TestUtils.adminClientSecurityConfigs(securityProtocol, trustStoreFile, clientSaslProperties)
-    securityProps.forEach { (key, value) => adminClientConfig.put(key.asInstanceOf[String], value) }
+    securityProps.forEach((key, value) => adminClientConfig.put(key.asInstanceOf[String], value))
```
Comment (chia7712) [inline core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala]: How about ```adminClientConfig.putAll(securityProps)```?

File: core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala [hunk 2]
```diff
@@ -32,17 +32,55 @@ import org.junit.jupiter.api.{AfterEach, BeforeEach, Test}
...
+    accessControlEntry(JaasTestUtils.KafkaClientPrincipalUnqualifiedName2, ALL, ALLOW))
+
+  val Topic2All = new AclBinding(
+    new ResourcePattern(ResourceType.TOPIC, Topic2, PatternType.LITERAL),
+    accessControlEntry(JaasTestUtils.KafkaClientPrincipalUnqualifiedName2, DELETE, ALLOW))
+
+  private def accessControlEntry(
+    userName: String,
+    operation: AclOperation,
+    permissionType: AclPermissionType
```
Comment (chia7712) [inline core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala]: It seems this filed is always ```ALLOW```

File: core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala [hunk 3]
```diff
@@ -32,17 +32,55 @@ import org.junit.jupiter.api.{AfterEach, BeforeEach, Test}
 
 import scala.jdk.CollectionConverters._
 
+object DescribeAuthorizedOperationsTest {
+  val Group1 = "group1"
```
Comment (chia7712) [inline core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala:36]: Is ```GROUP_1" more better? it is global variable in this test.

File: core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala [hunk 4]
```diff
@@ -73,11 +114,6 @@ class DescribeAuthorizedOperationsTest extends IntegrationTestHarness with SaslS
...
   }
 
-  private def accessControlEntry(userName: String, permissionType: AclPermissionType, operation: AclOperation): AccessControlEntry = {
-    new AccessControlEntry(new KafkaPrincipal(KafkaPrincipal.USER_TYPE, userName).toString,
-      AclEntry.WildcardHost, operation, permissionType)
-  }
-
   @AfterEach
   override def tearDown(): Unit = {
     if (client != null)
```
Comment (chia7712) [inline core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala]: unnecessary check

File: core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala [hunk 5]
```diff
@@ -32,17 +32,55 @@ import org.junit.jupiter.api.{AfterEach, BeforeEach, Test}
...
+}
+
 class DescribeAuthorizedOperationsTest extends IntegrationTestHarness with SaslSetup {
+  import DescribeAuthorizedOperationsTest._
+
   override val brokerCount = 1
   this.serverConfig.setProperty(KafkaConfig.ZkEnableSecureAclsProp, "true")
   this.serverConfig.setProperty(KafkaConfig.AuthorizerClassNameProp, classOf[AclAuthorizer].getName)
 
   var client: Admin = _
```
Comment (chia7712) [inline core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala:82]: all test cases need client so we should create it in construction or ```BeforeEach``` block.

**Review by chia7712 — APPROVED** [review]
@dajac Nice refactor. overall + 1

File: core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala (hunk 2 above)
Comment (dajac) [reply core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala]: Indeed, let me remove that field.

File: core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala (hunk 3 above)
Comment (dajac) [reply core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala:36]: In Scala, we tend to only capitalize the first letter of constants. At least, this how I have been doing it so far. I prefer to keep it as is.

File: core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala (hunk 4 above)
Comment (dajac) [reply core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala]: I think that it is better to keep it as `client` could not be initialized. For instance, let's say that there is an exception raised when the client is constructed. In this case, `client` would remain `null`.

**Comment by dajac** [comment]
@chia7712 Thanks for your review. I have addressed your comments. Do you mind having a second quick look at it?

File: core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala (hunk 4 above)
Comment (chia7712) [reply core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala]: Sorry for unclear comment. 

My point was ```Utils.closeQuietly``` includes null check so we don't need to add same check again.

Comment (dajac) [reply core/src/test/scala/integration/kafka/api/DescribeAuthorizedOperationsTest.scala]: Got it, thanks for the clarification. I have removed the null check.

**Review by chia7712 — APPROVED** [review]
@dajac +1 to nice patch!

**Comment by dajac** [comment]
The failed test is not related to this PR:
* Build / JDK 15 / org.apache.kafka.clients.consumer.internals.FetcherTest.testEarlierOffsetResetArrivesLate()

**Pull request merged**